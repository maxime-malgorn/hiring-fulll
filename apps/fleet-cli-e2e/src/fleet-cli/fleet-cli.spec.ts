import { execSync } from 'child_process';
import { join } from 'path';

describe('CLI tests', () => {
  const executeCommand = (args?: string): string => {
    const cliPath = join(process.cwd(), 'dist/apps/fleet-cli');
    const cliArguments = args ? ` ${args}` : '';
    return execSync(`node ${cliPath}${cliArguments}`).toString();
  };

  it('should create a fleet', () => {
    expect(executeCommand('create fleet-1')).toMatchSnapshot();
  });

  it('should register a vehicle', () => {
    const validCommand = 'register-vehicle fleet-1 AA-123-AA';

    // valid cases
    expect(executeCommand(validCommand)).toMatchSnapshot();

    // error cases
    expect(executeCommand(validCommand)).toMatchSnapshot('Already registered');
    expect(
      executeCommand('register-vehicle fleet-2 AA-123-AA')
    ).toMatchSnapshot();
    expect(
      executeCommand('register-vehicle fleet-1 invalid-plate')
    ).toMatchSnapshot();
  });

  it('should localize a vehicle', () => {
    // valid cases
    expect(
      executeCommand(`localize-vehicle fleet-1 AA-123-AA 23 87`)
    ).toMatchSnapshot();
    expect(
      executeCommand('localize-vehicle fleet-1 AA-123-AA 23 87 40')
    ).toMatchSnapshot();

    // error cases
    expect(
      executeCommand('localize-vehicle fleet-1 BB-123-BB 23 87')
    ).toMatchSnapshot();
    expect(
      executeCommand('localize-vehicle fleet-1 BB-123-BB 23 200')
    ).toMatchSnapshot();
    expect(
      executeCommand('localize-vehicle fleet-1 BB-123-BB 99 87')
    ).toMatchSnapshot();
  });
});
