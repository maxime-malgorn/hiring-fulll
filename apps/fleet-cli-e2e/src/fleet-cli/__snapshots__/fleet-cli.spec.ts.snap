// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI tests should create a fleet 1`] = `
"✔ Fleet created with id fleet-1
"
`;

exports[`CLI tests should localize a vehicle 1`] = `
"✔ Vehicle AA-123-AA parked at 23,87 with altitude not specified
"
`;

exports[`CLI tests should localize a vehicle 2`] = `
"✔ Vehicle AA-123-AA parked at 23,87 with altitude 40
"
`;

exports[`CLI tests should localize a vehicle 3`] = `
"❌ Cannot park vehicle (Error: Vehicle not found)
"
`;

exports[`CLI tests should localize a vehicle 4`] = `
"❌ Cannot park vehicle (Error: Vehicle not found)
"
`;

exports[`CLI tests should localize a vehicle 5`] = `
"❌ Cannot park vehicle (Error: Vehicle not found)
"
`;

exports[`CLI tests should register a vehicle 1`] = `
"✔ Vehicle AA-123-AA created and registered to fleet fleet-1
"
`;

exports[`CLI tests should register a vehicle 2`] = `
"❌ Cannot register vehicle to fleet (Error: Fleet not found)
"
`;

exports[`CLI tests should register a vehicle 3`] = `
"❌ Cannot register vehicle to fleet (Error: Invalid plate format)
"
`;

exports[`CLI tests should register a vehicle: Already registered 1`] = `
"❌ Cannot register vehicle to fleet (Error: Vehicle already exists in this fleet)
"
`;
